source/Calamari/Integration/Scripting/WindowsPowerShell/Bootstrap.ps1

```diff
@@ -12,46 +12,137 @@ $ErrorActionPreference = 'Stop'
 # Functions
 # -----------------------------------------------------------------
 
-function Write-VersionTable
+function Log-VersionTable
 {
 	Write-Verbose ($PSVersionTable | Out-String)
 }
 
+function Log-EnvironmentInformation
+{
+	if ($OctopusParameters.ContainsKey("Octopus.Action.Script.SuppressEnvironmentLogging")) {
+		if ($OctopusParameters["Octopus.Action.Script.SuppressEnvironmentLogging"] -eq "True") {
+			return;
+		}
+	}
+
+	Write-Host "##octopus[stdout-verbose]"
```

Comment:
@matt-richardson yeah, I've read http://www.jsnover.com/blog/2013/12/07/write-host-considered-harmful/ and had the same reaction. We've actually seen a couple of bugs around sensitive variable handling where PowerShell functions used `Write-Output` which affected the return value from the function in certain surprising circumstances, and took a long time to diagnose.

Since that time I've been quite deliberately using `Write-Host` where I only want a `Console.WriteLine()` equivalent - and so far, so good. :)

In this case I'd err on the side of caution and stick with `Write-Host` - also for consistency.